ntbs-service-unit-tests/Helpers/EpisodesExtensionMethodsTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ntbs_service.Helpers;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Enums;
+using Xunit;
+
+namespace ntbs_service_unit_tests.Helpers
+{
+    public class EpisodesExtensionMethodsTest
+    {
+        [Fact]
+        public void OrderForEpisodesCorrectlyOrdersEvents()
+        {
+            // Arrange
+            var earlierDate = new DateTime(2011, 1, 25);
+            var laterDate = new DateTime(2012, 12, 25);
+            var treatmentEvents = new List<TreatmentEvent>
+            {
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.TransferIn },
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.TransferOut },
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.DiagnosisMade },
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.TreatmentStart },
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.TreatmentOutcome },
+                new TreatmentEvent { EventDate = laterDate, TreatmentEventType = TreatmentEventType.TreatmentRestart },
+
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.TransferIn },
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.TransferOut },
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.DiagnosisMade },
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.TreatmentStart },
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.TreatmentRestart },
+                new TreatmentEvent { EventDate = earlierDate, TreatmentEventType = TreatmentEventType.TreatmentOutcome },
+            };
+            Random rng = new Random();
+            var randomisedOrderList = treatmentEvents.OrderBy(ev => rng.Next());
```

Comment:
It's generally considered a bad idea to include randomness in your tests in this way. The reason is that your test will be different every time, and so it's possible that it will pass most of the time, but only very occasionally fail, even when there is a bug.

As a general rule of thumb you should set out a selection of test cases that you are confident will only pass when the code is correct and only fail when there is a bug. In this instance, I think it would be enough to have 2 tests, one with the events starting in the expected order, and the other with them starting in the reverse order. I don't think there are any other edge cases that require particular attention here, but if there were then we should write a specific test for them, rather than hoping to find them via randomness.

Reply:
Yes that's understandable, I have now changed it to use the list as is and then use a reversed list.